tests/phpunit/src/DrupalAuthenticationManagerTest.php

```
@@ -415,10 +415,113 @@ private function createDriverManagerMock(): DrupalDriverManagerInterface {
+    $page->method('findButton')->with('Log in')->willReturn($submit);
+    // Simulate: logged_in_selector not found on first call, found on second.
+    $page->method('has')->willReturnCallback(function (string $selector, string $locator) use (&$callCount): bool {
+      if ($locator === 'body.logged-in') {
+        $callCount++;
+        // First two calls return FALSE (during wait loop and loggedIn check),
+        // then return TRUE.
+        return $callCount > 2;
+      }
+      return FALSE;
+    });
+    $page->method('find')->willReturnCallback(function (string $selector, string $locator) use ($page): ?DocumentElement {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$selector'. (undefined)

(UnusedFormalParameter)